### PR TITLE
ptr::addr_of documentation improvements

### PR DESCRIPTION
**Pull request by RalfJung**
While writing https://github.com/rust-lang/reference/pull/1001 I figured I could also improve the docs here a bit.

### CODE REVIEW

File: library/core/src/ptr/mod.rs [hunk 1]
```diff
@@ -1540,6 +1544,12 @@ fnptr_impls_args! { A, B, C, D, E, F, G, H, I, J, K, L }
 /// let raw_f2 = ptr::addr_of!(packed.f2);
 /// assert_eq!(unsafe { raw_f2.read_unaligned() }, 2);
 /// ```
+///
+/// See [`addr_of_mut`] for how to create a pointer to ininitialized data.
+/// Doing that with `addr_of` would not make much sense since one could only
+/// read the data, and that would be Undefined Behavior.
+///
+/// [`addr_of_mut`]: macro.addr_of_mut.html
```
Comment (JohnTitor) [inline library/core/src/ptr/mod.rs:1552]:
```suggestion
```

Thanks to the intra-doc links, we don't need these lines.

Comment (RalfJung) [reply library/core/src/ptr/mod.rs:1552]: I wasn't sure these would work with macros, and whether they required a trailing `!` or not... I guess we'll see.

File: library/core/src/ptr/mod.rs [hunk 2]
```diff
@@ -1540,6 +1544,10 @@ fnptr_impls_args! { A, B, C, D, E, F, G, H, I, J, K, L }
 /// let raw_f2 = ptr::addr_of!(packed.f2);
 /// assert_eq!(unsafe { raw_f2.read_unaligned() }, 2);
 /// ```
+///
+/// See [`addr_of_mut`] for how to create a pointer to ininitialized data.
```
Comment (kennytm) [inline library/core/src/ptr/mod.rs:1548]:
```suggestion
/// See [`addr_of_mut`] for how to create a pointer to unininitialized data.
```

Comment (RalfJung) [reply library/core/src/ptr/mod.rs:1548]: *oops* good catch!

**Pull request merged**